crates/starknet-os/src/hints/kzg.rs

```diff
@@ -0,0 +1,390 @@
+
+    let da_segment = range
+        .iter()
+        .map(|c| {
+            c.get_int().map(|i| i.to_bigint()).ok_or(HintError::CustomHint(
+                "DA segment must contain only integer entries".to_owned().into_boxed_str(),
+            ))
+        })
+        .collect::<Result<Vec<_>, _>>()?;
+    let commitments = polynomial_coefficients_to_kzg_commitment(da_segment)
+        .map_err(|e| HintError::CustomHint(e.to_string().into_boxed_str()))?;
+    let splits: Vec<MaybeRelocatable> = [commitments.0.into(), commitments.1.into()]
+        .into_iter()
+        .map(MaybeRelocatable::Int)
+        .collect::<Vec<MaybeRelocatable>>();
+    vm.write_arg(kzg_ptr, &splits)?;
+
+    Ok(())
+}
+
+pub const STORE_DA_SEGMENT: &str = indoc! {r#"import itertools
+
+from starkware.python.utils import blockify
+
+kzg_manager.store_da_segment(
+    da_segment=memory.get_range_as_ints(addr=ids.state_updates_start, size=ids.da_size)
+)
+kzg_commitments = [
+    kzg_manager.polynomial_coefficients_to_kzg_commitment_callback(chunk)
+    for chunk in blockify(kzg_manager.da_segment, chunk_size=ids.BLOB_LENGTH)
+]
+
+ids.n_blobs = len(kzg_commitments)
+ids.kzg_commitments = segments.add_temp_segment()
+ids.evals = segments.add_temp_segment()
+
+segments.write_arg(ids.kzg_commitments.address_, list(itertools.chain(*kzg_commitments)))"#};
+
+pub fn store_da_segment<PCS>(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError>
+where
+    PCS: PerContractStorage + 'static,
+{
+    execute_coroutine(store_da_segment_async::<PCS>(vm, exec_scopes, ids_data, ap_tracking, constants))?
+}
+
+pub async fn store_da_segment_async<PCS>(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError>
+where
+    PCS: PerContractStorage + 'static,
+{
+    let state_updates_start = get_ptr_from_var_name(vars::ids::STATE_UPDATES_START, vm, ids_data, ap_tracking)?;
+    let da_size = get_integer_from_var_name(vars::ids::DA_SIZE, vm, ids_data, ap_tracking)?.to_biguint();
+    let da_size: usize = da_size.try_into().map_err(|_| HintError::BigintToU32Fail)?;
+
+    let da_segment: Vec<Felt252> =
+        vm.get_integer_range(state_updates_start, da_size)?.into_iter().map(|s| *s).collect();
+
+    // TODO: "BLOB_LENGTH" doesn't appear to be in scope, although it's clear from the OS code that it's
+    // 4096 let blob_length = get_integer_from_var_name(vars::ids::BLOB_LENGTH, vm, ids_data,
+    // ap_tracking)?.to_biguint(); let blob_length = get_constant(vars::ids::BLOB_LENGTH,
+    // constants)?.to_biguint(); let blob_length: usize = blob_length.try_into().map_err(|_|
+    // HintError::BigintToU32Fail)?;
```

Comment:
Thanks, `cargo fmt`... this needs to be investigated (or, at a minimum this comment needs to be cleaned up).

Reply:
Cleaned up the comment in 6f36912. I'm still not sure why we can't find `BLOB_LENGTH` as a constant, it looks like the same case as `MERKLE_HEIGHT` in the Cairo code which we fetch via `get_constant()` in SNOS.